libs/utils/env.py

```diff
@@ -644,6 +644,17 @@ def _init_platform(self):
         # Adding topology information
         self.platform['topology'] = self.topology.get_level("cluster")
 
+        # Adding kernel build information
+        kver = self.target.execute('uname -r').strip()
+        match = KVERSION_RE.match(kver)
+        if match:
+            self.platform['kernel_version'] = match.group('ver')
+            self.platform['kernel_sha1'] = match.group('sha1')
+        self.platform['kernel_version_full'] = kver
+        self.platform['build'] = str(self.target.kernel_version)
```

Comment:
I suggest either putting all the kernel version stuff under a sub-dict keyed with "kernel" (`self.platform['kernel'] = {'build': foo, 'sha1': bar, ...}`) or changing '"build"' to '"kernel_build"'.